src/Storefront/Controller/ErrorController.php

```diff
@@ -66,25 +66,21 @@ public function error(\Throwable $exception, Request $request, SalesChannelConte
             $request->attributes->set('navigationId', $context->getSalesChannel()->getNavigationCategoryId());
 
             $salesChannelId = $context->getSalesChannel()->getId();
-            $cmsErrorLayoutId = $this->systemConfigService->getString('core.basicInformation.http404Page', $salesChannelId);
-            if ($cmsErrorLayoutId !== '' && $is404StatusCode) {
-                $errorPage = $this->errorPageLoader->load($cmsErrorLayoutId, $request, $context);
-
-                $response = $this->renderStorefront(
-                    '@Storefront/storefront/page/content/index.html.twig',
-                    ['page' => $errorPage]
-                );
-            } else {
-                $errorTemplate = $this->errorTemplateResolver->resolve($exception, $request);
-
-                if (!$request->isXmlHttpRequest()) {
-                    $header = $this->headerPageletLoader->load($request, $context);
-                    $errorTemplate->setHeader($header);
-                }
+            $cms404ErrorLayoutId = $this->systemConfigService->getString('core.basicInformation.http404Page', $salesChannelId);
 
-                $response = $this->renderStorefront($errorTemplate->getTemplateName(), ['page' => $errorTemplate]);
+            if ($cms404ErrorLayoutId !== '' ) {
```

Comment:
This if does not contain $is404StatusCode. Is there any special reason for that?